taskchampion/xtask/src/main.rs

```diff
@@ -29,3 +41,106 @@ fn codegen() -> anyhow::Result<()> {
+/// The argument "X.Y" will replace the text any place the MSRV is found.
+fn msrv(args: Vec<String>) -> anyhow::Result<()> {
+    if args.len() != 2 {
+        // check that (X.Y) argument is (mostly) valid:
+        if !args[2].chars().all(|c| c.is_numeric() || c == '.') {
+            anyhow::bail!("xtask: Invalid argument format. Xtask MSRV argument takes the form \"X.Y(y)\", where XYy are numbers. eg: `cargo run xtask MSRV 1.68`");
+        }
+
+        // for each file in const paths vector
+        for path in MSRV_FILE_PATHS {
+            let path: &Path = Path::new(&path);
+            let mut is_pattern_in_file = false;
+
+            // check if path exists, if not, skip then continue
+            if !path.exists() {
+                println!(
+                    "xtask: Warning: xtask: Path \"{}\" not found. Skipping.",
+                    path.display()
+                );
+                continue;
+            }
+
+            let mut file: File = File::options().read(true).write(true).open(path)?;
+            let reader = BufReader::new(&file);
+
+            // set the file_extension if it exists and is valid UTF-8
+            let mut file_extension = String::new();
+            if path.extension().is_some() && path.extension().unwrap().to_str().is_some() {
+                file_extension = path.extension().unwrap().to_str().unwrap().to_string();
+            } else {
+                anyhow::bail!(
+                    "xtask: {} does not have a valid extension and cannot be processed.",
+                    format!("{:#?}", path.as_os_str())
+                );
+            }
+
+            // set the comment glyph (#, //, --) pattern to search for based on file type
+            let mut comment_glyph = String::new();
+            match file_extension.as_str() {
+                "rs" => comment_glyph = "//".to_string(),
+                "toml" | "yaml" | "yml" => comment_glyph = "#".to_string(),
+                _ => anyhow::bail!("xtask: Support for file extension {} is not yet implemented in `cargo xtask msrv` command.", file_extension)
+            }
+
+            // set search string and the replacement string
+            let re = Regex::new(r#"(#|/{2,}) *MSRV *= *"*([0-9]+\.*)+"*.*"#).unwrap();
+            let replacement_string = format!("{} MSRV = \"{}\"", comment_glyph, args[2]);
+
+            // for each line in file
+            let mut file_string = String::new();
+            for line in reader.lines() {
+                let line_ref = &line.as_ref().unwrap();
+                let pattern_offset = re.find(line_ref);
+
+                // if a pre-existing MSRV pattern is found and is different, update it.
+                if pattern_offset.is_some() {
+                    if pattern_offset.unwrap().as_str() != replacement_string {
```

Comment:
This lint is pointing out that `if foo.is_some() { .. foo.unwrap() .. }` is better expressed as `if let Some(bar) = foo { .. bar .. }`.